classes/importer.php

```diff
@@ -0,0 +1,529 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace tiny_elements;
+
+use tiny_elements\local\constants;
+use tiny_elements\local\utils;
+use core\exception\moodle_exception;
+
+/**
+ * Class importer
+ *
+ * @package    tiny_elements
+ * @author     Stefan Hanauska <stefan.hanauska@csg-in.de>
+ * @copyright  2025 ISB Bayern
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class importer {
+    /** @var int $contextid */
+    protected int $contextid = 1;
```

Comment:
This is irritating. You are already setting it correctly in the constructor, so just do not initialize it here, so one does not have the (assumption) that it is set to 1.